smaily_for_drupal/src/Form/SubscribeForm.php

```diff
@@ -117,4 +120,38 @@ public function removeHiddenDrupalInputs(array $form) {
     return $form;
   }
 
+  /**
+   * Handle and display response messages from Smaily.
+   */
+  public function handleResponseMessage() {
+    $request = \Drupal::request();
```

Comment:
There isn't anything wrong with this code technically, but it can be simplified so you wouldn't need to access `query->all()` all the time.

```php
$params = \Drupal::request()->query->all();
$message = isset($params['message']) ? trim($params['message']) : NULL;
$code = isset($params['code']) ? (int) trim($params['code']) : NULL;

if ($message && $code) {
    // ...
}
```

And to reduce the **curly-bracket-pyramid-of-doom** you can always return early in a method:
```php
if (!$message && !$code) {
    return;
}

switch($code) {
    // ...
}
```